### PR TITLE
CMake: Fix test labels

### PR DESCRIPTION
**Pull request by marbre**
Prior to the changes all `_` in the path got replaced with `/`:

Before:
```
  Test  #30: iree/test/e2e/linalg/path:pw_add.mlir.test
  Test  #31: iree/test/e2e/linalg/path:pw_add_mul.mlir.test
  Test  #32: iree/test/e2e/linalg/path:pw_add_multiwg.mlir.test
  Test  #33: iree/test/e2e/linalg/path:reduce.mlir.test

  Test #268: iree/samples/custom/modules:custom_modules_test
  Test #269: iree/samples/custom/modules/dialect/test:conversion.mlir.test
  Test #270: iree/samples/custom/modules/dialect/test:custom_ops.mlir.test
  Test #271: iree/samples/simple/embedding:simple_embedding_test
```

After:
```
  Test  #30: iree/test/e2e/linalg_path:pw_add.mlir.test
  Test  #31: iree/test/e2e/linalg_path:pw_add_mul.mlir.test
  Test  #32: iree/test/e2e/linalg_path:pw_add_multiwg.mlir.test
  Test  #33: iree/test/e2e/linalg_path:reduce.mlir.test

  Test #268: iree/samples/custom_modules:custom_modules_test
  Test #269: iree/samples/custom_modules/dialect/test:conversion.mlir.test
  Test #270: iree/samples/custom_modules/dialect/test:custom_ops.mlir.test
  Test #271: iree/samples/simple_embedding:simple_embedding_test
```

### CODE REVIEW

File: build_tools/cmake/iree_cc_test.cmake [hunk 1]
```diff
@@ -108,7 +108,8 @@ function(iree_cc_test)
   # We run all our tests through a custom test runner to allow temp directory
   # cleanup upon test completion.
 
-  string(REPLACE "_" "/" _PACKAGE_PATH ${_PACKAGE_NAME})
+  iree_package_ns(_PACKAGE_NS)
```
Comment (GMNGeoffrey) [inline build_tools/cmake/iree_cc_test.cmake:111]: _PACKAGE_NS is defined with iree_package_ns already above. Can you reuse it instead?

File: build_tools/cmake/iree_lit_test.cmake [hunk 2]
```diff
@@ -59,7 +59,8 @@ function(iree_lit_test)
   endforeach(_DATA_DEP)
 
   # We run all our tests through a custom test runner to allow setup and teardown.
```
Comment (GMNGeoffrey) [inline build_tools/cmake/iree_lit_test.cmake:61]: This comment is drifting further from the thing it's explaining. Can you shift it down? Actually can it go directly above run_test? CMake doesn't get upset if there's a comment in the function call, does it?

File: build_tools/cmake/iree_cc_test.cmake (hunk 1 above)
Comment (marbre) [reply build_tools/cmake/iree_cc_test.cmake:111]: Oh yeah, thanks! Missed that one.

File: build_tools/cmake/iree_lit_test.cmake (hunk 2 above)
Comment (marbre) [reply build_tools/cmake/iree_lit_test.cmake]: Didn't build locally, but the cmake configuration does not struggle with it. Let's check with the CI :) Further moved a similar comment in `iree_cc_test`.

**Pull request closed without merging**